src/main/scanLogic/scanManager.ts

```diff
@@ -32,25 +41,110 @@ export class ScanManager implements ExtensionComponent {
+            progress.report({ message: 'Get outdated resources' });
+            let resources: Resource[] = await this.getOutdatedResources();
+            if (resources.length === 0) {
+                return;
+            }
+            let progressManager: StepProgress = new StepProgress(progress);
+            progressManager.startStep('Update resources', resources.length);
+            this._logManager.logMessage(
+                'Updating outdated resources (' + resources.length + '): ' + resources.map(resource => resource.name).join(),
+                'INFO'
+            );
+            let updatePromises: Promise<any>[] = [];
+            resources.forEach(async resource =>
+                updatePromises.push(
+                    resource
+                        .update()
+                        .catch(err => {
+                            this._logManager.logError(<Error>err);
+                            result = false;
+                        })
+                        .finally(() => progressManager.reportProgress())
+                )
+            );
+            await Promise.all(updatePromises);
+        });
+        this._logManager.logMessage(
+            'Updating outdated extension resources finished ' + (result ? 'successfully' : 'with error'),
+            result ? 'INFO' : 'ERR'
+        );
+        return result;
+    }
+
+    private async getOutdatedResources(): Promise<Resource[]> {
+        if (!this.shouldCheckOutdated()) {
+            return [];
+        }
+        this._logManager.logMessage('Checking for outdated resources', 'INFO');
+        ScanManager.lastOutdatedCheck = Date.now();
+        let promises: Promise<boolean>[] = [];
+        let outdatedResources: Resource[] = [];
+        for (const resource of await this.getResources()) {
+            promises.push(
+                resource
+                    .isOutdated()
+                    .then(outdated => {
+                        if (outdated) {
+                            outdatedResources.push(resource);
+                        }
+                        return outdated;
+                    })
+                    .catch(err => {
+                        this._logManager.logError(<Error>err);
+                        return false;
+                    })
+            );
+        }
+        await Promise.all(promises);
+        return outdatedResources;
+    }
+
+    private shouldCheckOutdated(): boolean {
+        return !ScanManager.lastOutdatedCheck || Date.now() - ScanManager.lastOutdatedCheck > ScanManager.RESOURCE_CHECK_UPDATE_INTERVAL;
+    }
+
+    private async getResources(): Promise<Resource[]> {
+        let resources: Resource[] = [];
+        if (await this.isAnalyzerManagerSupported()) {
+            resources.push(BinaryRunner.getAnalyzerManagerResource(this._logManager));
+        } else {
+            this.logManager.logMessage('You are not entitled to run contextual analysis scans', 'WARN');
```

Comment:
This error shouldn't appear to the user. Can we make it DEBUG?

Reply:
done